account_invoice_inter_company/models/res_config.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+from odoo import models, fields, api
+
+
+class InterCompanyRulesConfig(models.TransientModel):
+
+    _inherit = 'base.config.settings'
+
+    company_id = fields.Many2one(
+        'res.company', string='Select Company',
+        help='Select company to setup Inter company rules.')
+    invoice_auto_validation = fields.Boolean(
```

Comment:
Add in `invoice_auto_validation` field the following line:
`default=lambda self: self.env.user.company_id.invoice_auto_validation,`

and then:

- remove field `company_id`, because you don't need it anymore
- remove method `def onchange_invoice_company_id()`
- remove method `def set_invoice_inter_company_configuration()`

This way you will also increase the coverage without adding extra tests.